### PR TITLE
MAINT: remove use of deprecated scipy.stats.ss helper function.

### PR DESCRIPTION
**Pull request by rgommers**
Closes gh-2424

Also some other minor cleanups and an addition to .gitignore that was needed.

### CODE REVIEW

**Comment by josef-pkt** [comment]
@rgommers Thanks
I see one more in sandbox/tsa/garch.py  (which is essentially a dead module now)

I haven't seen _statespace.pyx because I have to many uncommitted files that show up as untracked.

**Comment by rgommers** [comment]
indeed, thanks. not sure why I missed that one. ``ss`` is not the easiest to search for:)

File: statsmodels/regression/linear_model.py [hunk 1]
```diff
@@ -1494,7 +1492,6 @@ def compare_lm_test(self, restricted, demean=True, use_lr=False):
         elif cov_type == 'HAC':
             print("HAC")
             maxlags = self.cov_kwds['maxlags']
-            use_correction = self.cov_kwds['use_correction']
```
Comment (josef-pkt) [inline statsmodels/regression/linear_model.py]: I assume this is pylint/pyflakes cleaning. might be copy-paste extras

**Comment by josef-pkt** [comment]
Thanks Ralf, I'll merge when TravisCI comes back green.

**Comment by josef-pkt** [comment]
merged

File: statsmodels/regression/linear_model.py (hunk 1 above)
Comment (rgommers) [inline statsmodels/regression/linear_model.py]: indeed, pyflakes

**Pull request merged**